Update tests

medecau opened the pull request:
This pull-request removes python 3.1 and 3.2 from tox config.
Work is being made on python 3.4 using a different branch and 3.1 and 3.2 are outdated.

Also included are changes that add a new environment to tox config specific to running flake8.

I also made some changes to fix the warnings generated by flake8.

homu commented:
:umbrella: The latest upstream changes (presumably fc83236) made this pull request unmergeable. Please resolve the merge conflicts.

medecau commented:
I have allowed edits from maintainers if you care to make changes.

medecau commented:
Fixed merge conflicts.

Three tests are failing one is a timeout at build time the two others are timeouts during tests.

 - [169.35]https://travis-ci.org/0rpc/zerorpc-python/jobs/174521253
 - [169.43](https://travis-ci.org/0rpc/zerorpc-python/jobs/174521261)
 - [169.46](https://travis-ci.org/0rpc/zerorpc-python/jobs/174521264)

bombela commented on .travis.yml:
```diff
@@ -1,30 +1,77 @@
 sudo: false
 language: python
-python:
```
why removing this matrix configuration and replacing by a hand-generated version?

medecau replied:
because that matrix controls the travis systems python

the changes I have submitted set travis to use tox thus we don't need to care about different python versions at the system level as those will be controlled by tox inside it's virtual environments.

I am aware this makes travis config highly coupled to the tox config. I wanted to enforce the maintenance of both configs together.

bombela commented on tox.ini:
```diff
@@ -1,14 +1,43 @@
-    flake8 zerorpc bin
-    nosetests -v
-passenv = ZPC_TEST_TIME_FACTOR
+
```
why do you need to write every single combination by hand?

medecau replied:
because of the requirements?

if I were to permutate all sub versions of pyzmq I would not be able to set the requirement for each one

if you are aware of a better way to right this please submit the changes. As a maintainer of zerorpc-python you have access to `medecau:update-tests`

bombela commented:
I am sorry, but unless you can avoid writing by hand all the permutations, I will not merge this pull request. This is way too hard to maintain.

medecau commented:
- removed most of the TOXENV definitions on the travis file.
 - merged updates from master
 - added python 3.6 and pyzmq 16

@bombela 
do you think it would be okay to reduce the tests further? I'm thinking that python < 2.7 and python <3.6 could do with being tested on the latest pyzmq only and then test python 2.7 and python 3.6 and the major pyzmq versions. It would look like:

- py26-pyzmq
- py34-pyzmq
- py35-pyzmq
- py27-pyzmq
- py27-pyzmq13
- py27-pyzmq14
- py27-pyzmq15
- py27-pyzmq16
- py36-pyzmq
- py36-pyzmq13
- py36-pyzmq14
- py36-pyzmq15
- py36-pyzmq16
- flakes

medecau commented:
Travis is complaining about testing python 3.6 environments under python 3.5. See: https://github.com/travis-ci/travis-ci/issues/4794
